File: Data Structures & Algorithms/three-integer-sum/submission-8.go
import "slices"

func threeSum(nums []int) [][]int {
	res := [][]int{}
	slices.Sort(nums)
	for i := 0; i < len(nums) - 2; i++ {
		if i > 0 && nums[i] == nums[i-1] {
			continue
		}
		if nums[i] > 0 {
			break
		}
		target := -nums[i]
		for j, k := i + 1, len(nums) - 1; j < k; {
			sum := nums[j] + nums[k]
			if sum < target {
				j++
			} else if sum > target {
				k--
			} else {
				res = append(res, []int{nums[i],nums[j],nums[k]})
				j++
				for j < k && nums[j] == nums[j-1] {
					j++
				}
			}
		}
	}
	return res
}

File: Data Structures & Algorithms/binary-search/submission-8.cs
public class Solution {
    public int Search(int[] nums, int target) {
        if (nums.Length == 1)
        {
            return nums[0] == target ? 0 : -1;
        }

        int l = 0;
        int r = nums.Length - 1;

        while (l <= r)
        {
            int mid = l + ((r - l) / 2);
            
            if (nums[mid] == target)
            {
                return mid;
            }
            else if (nums[mid] < target)
            {
                l = mid + 1;
            }
            else
            {
                r = mid - 1;
            }
        }

        return -1;
    }
}
